src/utils/bind.ts

```diff
@@ -20,11 +23,11 @@ function _bind(router: Router, container: Container, identifier: symbol, _baseBe
   for (const route of routes) {
     const routeBeforeMiddlewares: Handler[] = (Reflect.getMetadata(MetadataKey.ROUTE_BEFORE_MIDDLEWARES, controller, route.handlerName) || [])
       .map((item: { isFactory: boolean, factoryOrHandler: (() => Handler) | Handler }) => {
-        return item.isFactory ? item.factoryOrHandler.call(controller) : item.factoryOrHandler;
+        return item.isFactory ? item.factoryOrHandler.call(controller, undefined as unknown as Request, undefined as unknown as Response, undefined as unknown as NextFunction) : item.factoryOrHandler; // TODO Fix parameters
```

Comment:
@SylvainEstevez had no idea how to make the compiler happy... this for sure isn't the right solution :/

Reply:
```typescript
@beforeFactory(function (this: DummyController) {
  return async (req: Request, res: Response, next: NextFunction) {
    // Do stuff
  }
})
@before(async (req: Request, res: Response, next: NextFunction) => {
  // Do stuff
})
class DummyController {

}
```

In the case of `beforeFactory`, `isFactory` will be `true` and no arguments are expected. We simply make sure sure to bind `this` to the controller so that it's properties may be accessed by the middleware.

```typescript
(item: { isFactory: true, factoryOrHandler: (() => Handler) } | { isFactory: false, factoryOrHandler: Handler }) => {
  return item.isFactory ? item.factoryOrHandler.call(controller) : item.factoryOrHandler;
}
```

Should make the compiler happy?

Reply:
Works <3